### ISSUE TITLE
[Regression] [Security]-ish related: Quartz Jobstore doesn't work with secure transport

### ISSUE THREAD

**Issue by exabrial** [metabase/metabase#9650]
We're using Percona MySQL  8.0.15-5 and Metabase 0.32.1. MySQL is set to reject unencrypted connections to prevent accidents. Previous version of Metabase operated without problems. 

EDIT:
It's important to note that our Metabase datastore is _not_ embedded HSQLDB, but connects to MySQL for its metadata. 

Here's the environment file for Metabase:
```
MB_DB_CONNECTION_URI=mysql://xxx.com:3306/metabase?user=metabase&password=totallyNotOurPasswordForReal&useSSL=true
MB_ENCRYPTION_SECRET_KEY=abcdefghi=
```

And system.d unit:
```
[Unit]
Description=Metabase Service
After=network.target

[Service]
EnvironmentFile=/etc/default/metabase
Type=simple
WorkingDirectory=/opt/metabase
ExecStart=/usr/bin/java -Xmx1280M -jar /opt/metabase/metabase.jar
User=metabase
Group=metabase
RestartSec=10
Restart=always
StandardOutput=syslog
StandardError=syslog
SyslogIdentifier=metabase

[Install]
WantedBy=multi-user.target
```

We're seeing this exception:
```
Apr  1 09:39:41 metabase metabase[5862]: 04-01 09:39:41 #033[1mERROR jdbcjobstore.JobStoreTX#033[0m :: ClusterManager: Error managing cluster: Failed to obtain DB connection from data source 'db': java.sql.SQLException: Connections could not be acquired from the underlying database!
Apr  1 09:39:41 metabase metabase[5862]: org.quartz.JobPersistenceException: Failed to obtain DB connection from data source 'db': java.sql.SQLException: Connections could not be acquired from the underlying database! [See nested exception: java.sql.SQLException: Connections could not be acquired from the underlying database!]
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreSupport.getConnection(JobStoreSupport.java:767)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreTX.getNonManagedTXConnection(JobStoreTX.java:71)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreSupport.doCheckin(JobStoreSupport.java:3232)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreSupport$ClusterManager.manage(JobStoreSupport.java:3857)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreSupport$ClusterManager.run(JobStoreSupport.java:3894)
Apr  1 09:39:41 metabase metabase[5862]: Caused by: java.sql.SQLException: Connections could not be acquired from the underlying database!
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.sql.SqlUtils.toSQLException(SqlUtils.java:118)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutPooledConnection(C3P0PooledConnectionPool.java:692)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.impl.AbstractPoolBackedDataSource.getConnection(AbstractPoolBackedDataSource.java:140)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.utils.PoolingConnectionProvider.getConnection(PoolingConnectionProvider.java:247)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.utils.DBConnectionManager.getConnection(DBConnectionManager.java:108)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.quartz.impl.jdbcjobstore.JobStoreSupport.getConnection(JobStoreSupport.java:764)
Apr  1 09:39:41 metabase metabase[5862]: #011... 4 more
Apr  1 09:39:41 metabase metabase[5862]: Caused by: com.mchange.v2.resourcepool.CannotAcquireResourceException: A ResourcePool could not acquire a resource from its primary factory or source.
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.awaitAvailable(BasicResourcePool.java:1469)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.prelimCheckoutResource(BasicResourcePool.java:644)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.checkoutResource(BasicResourcePool.java:554)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutAndMarkConnectionInUse(C3P0PooledConnectionPool.java:758)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool.checkoutPooledConnection(C3P0PooledConnectionPool.java:685)
Apr  1 09:39:41 metabase metabase[5862]: #011... 8 more
Apr  1 09:39:41 metabase metabase[5862]: Caused by: java.sql.SQLException: Connections using insecure transport are prohibited while --require_secure_transport=ON.
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.get(ExceptionMapper.java:255)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.util.exceptions.ExceptionMapper.getException(ExceptionMapper.java:165)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1199)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.util.Utils.retrieveProxy(Utils.java:560)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.MariaDbConnection.newConnection(MariaDbConnection.java:174)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.Driver.connect(Driver.java:92)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.DriverManagerDataSource.getConnection(DriverManagerDataSource.java:175)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.WrapperConnectionPoolDataSource.getPooledConnection(WrapperConnectionPoolDataSource.java:220)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.WrapperConnectionPoolDataSource.getPooledConnection(WrapperConnectionPoolDataSource.java:206)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.c3p0.impl.C3P0PooledConnectionPool$1PooledConnectionResourcePoolManager.acquireResource(C3P0PooledConnectionPool.java:203)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.doAcquire(BasicResourcePool.java:1138)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.doAcquireAndDecrementPendingAcquiresWithinLockOnSuccess(BasicResourcePool.java:1125)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool.access$700(BasicResourcePool.java:44)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.resourcepool.BasicResourcePool$ScatteredAcquireTask.run(BasicResourcePool.java:1870)
Apr  1 09:39:41 metabase metabase[5862]: #011at com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:696)
Apr  1 09:39:41 metabase metabase[5862]: Caused by: java.sql.SQLException: Connections using insecure transport are prohibited while --require_secure_transport=ON.
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.authentication(AbstractConnectProtocol.java:934)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.handleConnectionPhases(AbstractConnectProtocol.java:850)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connect(AbstractConnectProtocol.java:507)
Apr  1 09:39:41 metabase metabase[5862]: #011at org.mariadb.jdbc.internal.protocol.AbstractConnectProtocol.connectWithoutProxy(AbstractConnectProtocol.java:1195)
Apr  1 09:39:41 metabase metabase[5862]: #011... 12 more
```

**Comment by exabrial** [comment]
Also, running on JDK 1.8:
```
$ java -version
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
```

**Comment by flamber** [comment]
@exabrial Did you try 0.32.0? Because #9631 was added to 0.32.1 to fix issues with MySQL SSL, so it would be interesting to see if that PR caused your issue.

**Comment by exabrial** [comment]
Dumb question, where are the older release jars at?

**Comment by flamber** [comment]
You should be able to use the `tag` name to modify the download URL for any older versions:
https://downloads.metabase.com/v0.32.0/metabase.jar

**Comment by exabrial** [comment]
It looks like the same issue exists in v0.32.0

**Comment by exabrial** [comment]
I don't understand Clojure well know to know what's going on here, but this might be the problem: https://github.com/metabase/metabase/commit/5324041233352031d1ead61eff140a71faad5078#diff-e1e479d12db45f1276f67b0eb6689dc6R107

**Comment by camsaul** [comment]
@exabrial that line is just a style change. Doesn't affect anything else.

It sounds like it's still trying to use an insecure connection

> Caused by: java.sql.SQLException: Connections using insecure transport are prohibited while --require_secure_transport=ON.

I'll take a look.

**Comment by camsaul** [comment]
Ok. I've identified the problem. Connection properties such as `useSSL=true` aren't getting passed to Quartz correctly because of changes in 0.32.0. I'll can fix this

**Comment by camsaul** [comment]
Fixed by #9671
